File: TestSample/python/src/world.py
from src.person import Person


class World:
    def __init__(self) -> None:
        self._people: list[Person] = []

    def add(self, person: Person) -> None:
        self._people.append(person)

    def hello_all(self) -> list[str]:
        return [p.hello() for p in self._people]

File: TestSample/rust/src/world.rs
use crate::person::Person;

pub struct World {
    people: Vec<Box<dyn Person>>,
}

impl World {
    pub fn new() -> Self {
        Self { people: Vec::new() }
    }

    pub fn add(&mut self, person: Box<dyn Person>) {
        self.people.push(person);
    }

    pub fn hello_all(&self) -> Vec<String> {
        self.people.iter().map(|p| p.hello()).collect()
    }
}
